apps/codeforafrica/src/lib/data/common/index.js

```diff
@@ -44,11 +44,113 @@ function getPageSlug({ params }) {
   return params?.slugs?.[pageSlugIndex] || "index";
 }
 
+function getDefaultErrorPageProps(statusCode = 404) {
+  if (statusCode === 500) {
+    return {
+      title: "Server Error. ",
+      subtitle: [
+        {
+          children: [
+            {
+              text: "Don't worry!, you can head back to our ",
+              children: null,
+            },
+            {
+              type: "link",
+              newTab: false,
+              url: "/",
+              children: [
+                {
+                  text: "homepage",
+                  children: null,
+                },
+              ],
+              href: "/",
+            },
+            {
+              text: "check out our most recent ",
+              children: null,
+            },
+            {
+              type: "link",
+              newTab: false,
+              url: "/projects",
+              children: [
+                {
+                  text: "projects",
+                  children: null,
+                },
+              ],
+              href: "/projects",
+            },
+            {
+              text: ", or read below some of the contents produced by our amazing team while the technical team is working on fixing the issue.",
+              children: null,
+            },
+          ],
+        },
+      ],
+    };
+  }
+
+  return {
+    title: "Whoops! This page got lost in conversation! ",
+    subtitle: [
+      {
+        children: [
+          {
+            text: "Don't worry!, you can head back to our ",
+            children: null,
+          },
+          {
+            type: "link",
+            newTab: false,
+            url: "/",
+            children: [
+              {
+                text: "homepage",
+                children: null,
+              },
+            ],
+            href: "/",
+          },
+          {
+            text: "check out our most recent ",
+            children: null,
+          },
+          {
+            type: "link",
+            newTab: false,
+            url: "/projects",
+            children: [
+              {
+                text: "projects",
+                children: null,
+              },
+            ],
+            href: "/projects",
+          },
+          {
+            text: ", or read below some of the contents produced by our amazing team.",
+            children: null,
+          },
+        ],
+      },
+    ],
+  };
+}
+
 export async function getPageProps(api, context) {
   const slug = getPageSlug(context);
   const {
     docs: [page],
   } = await api.findPage(slug);
+  if (!page) {
+    if (slug === "500") {
+      return getDefaultErrorPageProps(500);
+    }
+    return getDefaultErrorPageProps();
+  }
   if (!page) {
     return null;
   }
```

Comment:
```suggestion
    if (["404", "500"].includes(slug)) {
      return getDefaultErrorPageProps(slug);
    }
    return null;
  }
```

A single `if (!page)` should be enough.